frontend/app/(tabs)/home.tsx

```diff
@@ -72,14 +62,42 @@ const Index = () => {
     } finally {
       setIsLoading(false);
     }
-  }, [page, isLoading]);
+  }, [page, isLoading, searchText]);
+
+  const handleSearch = (text: string) => {
+    setSearchText(text);
+
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
+
+    searchTimeout.current = setTimeout(() => {
+      if (text.trim().length >= 2) {
+        loadInitialCards(text);
+      } else if (text.trim() === '') {
+        loadInitialCards();
+      }
+    }, 500);
+  };
+
+  const handleSearchSubmit = () => {
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
 
-  const handleSearch = (searchText: string) => {
-    setSearchText(searchText);
+    if (searchText.trim().length >= 2) {
+      loadInitialCards(searchText);
+    } else if (searchText.trim() === '') {
+      loadInitialCards();
+    }
```

Comment:
This looks very much like the code above, I think your agent is messing with you, you have to be more strict with it and also spot this changes when you go through the PR yourself.